### DISCUSSION TITLE
Update an unknown table based on inheritance

### DISCUSSION THREAD

**Discussion by ArthurValadares** [simolus3/drift#2930] [Q&A]
Hi there.

I would like to know if, using Drift, it is possible to update a row from an unknown table based on the table it inherits from. That is, let A be an abstract table with attributes a1, a2, a3, ... from which B and C inherit all attributes, is it possible to write a generic method that allows updating B or C based on the argument of the past type? I face this problem considering that many of my tables have common attributes, so I prefer to use inheritance to avoid repetition.

**Comment by simolus3** [comment]
You can always use the [runtime schema introspection](https://drift.simonbinder.eu/docs/dart-api/schema_inspection/) APIs for this. I think you can even get some type safety here by writing a generic method like this (where `A` is the name of your common table):

```dart
Future<void> updateCommonField<T extends TableInfo<A, D>, D>(T table,) {
  // ...
}
```

That allows you to call `table.asDslTable` go view the actual table as an instance of `A`, which might be useful for building queries. However, if `A` is an abstract table, it doesn't have a companion or a data class, I think you can't update columns in a type-safe manner.
